src/samplers/activelearner.py

```diff
@@ -0,0 +1,318 @@
+        metrics (list): List to store metrics.
+    """
+
+    sampler_interface = S.ACTIVE
+
+    def __init__(
+        self,
+        total_budget: int,
+        parser_kwargs,
+        model_kwargs,
+        train_kwargs,
+        *args,
+        **kwargs,
+    ):
+
+        # fmt: off
+        self.total_budget          = total_budget
+        self.bounds                = kwargs.get('bounds', [None])
+        self.parameters            = kwargs.get('parameters', ['NA']*len(self.bounds))
+        self.aquisition_batch_size = kwargs.get('aquisition_batch_size', 512)
+        self.init_num_samples      = kwargs.get('num_initial_points', self.aquisition_batch_size)
+        self.selection_method      = kwargs.get('selection_method', 'random')
+        self.kernel_transform      = kwargs.get('kernel_transform', [])
+        self.base_kernel           = kwargs.get('base_kernel', 'll')
+
+        self.parser_kwargs         = parser_kwargs
+        self.model_kwargs          = model_kwargs
+        self.train_kwargs          = train_kwargs
+        self.metrics = []
+
+        self.parser                = getattr(parsers, parser_kwargs.pop('type'))(**parser_kwargs)
+        # fmt: on
+
+    def get_initial_parameters(self):
+        """
+        Generates initial parameter samples.
+
+        Returns:
+            list: List of dictionaries containing initial parameter samples.
+
+        """
+        # TODO: more than just random samples
+        assert self.parameters[0] != "NA"
+        batch_samples = []
+        for _ in range(self.init_num_samples):
+            params = [
+                torch.distributions.Uniform(lb, ub).sample().item()
+                for (lb, ub) in self.bounds
+            ]
+            param_dict = dict(zip(self.parameters, params))
+            batch_samples.append(param_dict)
+        return batch_samples
+
+    def _get_new_idxs_from_pool(self, model, train, pool) -> torch.Tensor:
+        """
+        Selects new indices from the pool using BMDAL's select_batch function.
+
+        Args:
+            model: PyTorch model.
+            train: Training data.
+            pool: Pool data.
+
+        Returns:
+            torch.Tensor: Tensor containing new indices.
+
+        """
+        y_train = train[:, self.parser.output_col_idxs].float()
+        train = train[:, self.parser.input_col_idxs].float()
+        pool = pool[:, self.parser.input_col_idxs].float()
+        train_data = TensorFeatureData(train)
+        pool_data = TensorFeatureData(pool)
+
+        new_idxs, _ = select_batch(
+            batch_size=self.aquisition_batch_size,
+            models=[model],
+            data={"train": train_data, "pool": pool_data},
+            y_train=y_train,
+            selection_method=self.selection_method,
+            sel_with_train=True,
+            base_kernel=self.base_kernel,
+            kernel_transforms=self.kernel_transform,
+        )
+
+        return new_idxs
+
+    def get_next_parameter(
```

Comment:
I think it might be important to specify in the docstring that the pool and train data are expected to be normalised.